rpcserver.go

```diff
@@ -5766,15 +5766,38 @@ func (r *rpcServer) GetTransactions(ctx context.Context,
 	// To remain backwards compatible with the old api, default to the
 	// special case end height which will return transactions from the start
 	// height until the chain tip, including unconfirmed transactions.
-	var endHeight = btcwallet.UnconfirmedHeight
-
+	var (
+		endHeight  = btcwallet.UnconfirmedHeight
+		filterFunc lnwallet.FilterTxFn
+		txHashes   = make([]*chainhash.Hash, len(req.FilterTxHash))
+		err        error
+	)
 	// If the user has provided an end height, we overwrite our default.
 	if req.EndHeight != 0 {
 		endHeight = req.EndHeight
 	}
+	if len(req.FilterTxHash) > 0 {
+		for i := range req.FilterTxHash {
+			txHashes[i], err = chainhash.NewHashFromStr(
+				req.FilterTxHash[i],
+			)
+			if err != nil {
+				return nil, err
+			}
+		}
+		filterFunc = func(txs *lnwallet.TransactionDetail) bool {
```

Comment:
Hmm, this is pretty inefficient, we already have the `TxDetails` call which takes a specific `txid` and can query the index on disk directly: https://pkg.go.dev/github.com/btcsuite/btcwallet/wtxmgr#Store.TxDetails

Reply:
It is. The first version of this PR did implement wtxmgr to fetch transactions, but we simplyfied the code by adding a filter.